pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -42,28 +41,113 @@ type VolumeZone struct {
 }
 
 var _ framework.FilterPlugin = &VolumeZone{}
+var _ framework.PreFilterPlugin = &VolumeZone{}
 var _ framework.EnqueueExtensions = &VolumeZone{}
 
 const (
 	// Name is the name of the plugin used in the plugin registry and configurations.
 	Name = names.VolumeZone
 
+	stateKey framework.StateKey = Name
+
 	// ErrReasonConflict is used for NoVolumeZoneConflict predicate error.
 	ErrReasonConflict = "node(s) had no available volume zone"
 )
 
-var volumeZoneLabels = sets.NewString(
+// the state is initialized in PreFilter phase. because we save the pointer in
+// framework.CycleState, in the later phases we don't need to call Write method
+// to update the value
+type stateData struct {
+	// pvByVolume holds the pv information corresponding to pod's volume
+	// it's initialized in the PreFilter phase
+	pvByVolume map[string]*v1.PersistentVolume
```

Comment:
since the order of the pod's volumes doesn't change, this could be a slice

Reply:
Since we'll skip unbound volumes and volume without pvc, although the order doesn't change, the len of this slice will change. How do we handle this situation. Append a nil for those invalid pv so the index of the entire slice remains unchanged?